packages/core/src/components/forms/numericInput.tsx

```diff
@@ -317,9 +328,16 @@ export class NumericInput extends AbstractPureComponent2<HTMLInputProps & INumer
         const didMinChange = this.props.min !== prevProps.min;
         const didMaxChange = this.props.max !== prevProps.max;
         const didBoundsChange = didMinChange || didMaxChange;
-        if (didBoundsChange && this.state.value !== prevState.value) {
+        const didLocaleChange = this.props.locale !== prevProps.locale;
+        const didValueChange = this.state.value !== prevState.value;
+
+        if ((didBoundsChange && didValueChange) || didLocaleChange) {
             // we clamped the value due to a bounds change, so we should fire the change callback
-            this.props.onValueChange?.(+this.state.value, this.state.value, this.inputElement);
+            const valueToParse = didLocaleChange ? prevState.value : this.state.value;
```

Comment:
can we tweak this behavior to add special handling of the empty value case?

here, after switching the locale, I don't expect the input value to be set to `0`. it should remain undefined and continue showing the placeholder text:

![2020-11-16 15 39 45](https://user-images.githubusercontent.com/723999/99305768-3b128280-2822-11eb-93d6-8b8d18b62c8f.gif)

Reply:
ok, but I think we want to convert the number if the value is set

Reply:
> ok, but I think we want to convert the number if the value is set

yes